Sources/Extraction/MinutiaeDetection/MinutiaeDetection.c

```diff
@@ -0,0 +1,165 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include "MinutiaeDetection.h"
+#include "General/BinaryMap.h"
+#include "General/List.h"
+#include "General/Array.h"
+#include "Extraction/Model/Minutia.h"
+#include "Extraction/Model/Ridge.h"
+
+UInt8Array2D minutiaeLocations;
+
+static int CountNeighbors(int x, int y, BinaryMap* image) {
+  const int imageWidth = image->width;
+  const int imageHeight = image->height;
+  int acc = 0;
+
+  if ( x > 0 )                                 acc += BinaryMap_GetBit(image, x-1, y);
+  if ( x > 0 && y < imageHeight-1 )            acc += BinaryMap_GetBit(image, x-1, y+1);
+  if ( y <  imageHeight-1 )                    acc += BinaryMap_GetBit(image, x, y+1);
+  if ( x < imageWidth-1 && y < imageHeight-1 ) acc += BinaryMap_GetBit(image, x+1, y+1);
+  if ( x < imageWidth-1 )                      acc += BinaryMap_GetBit(image, x+1, y);
+  if ( x < imageWidth-1 && y > 0 )             acc += BinaryMap_GetBit(image, x+1, y-1);
+  if ( y > 0 )                                 acc += BinaryMap_GetBit(image, x, y-1);
+  if ( x > 0 && y > 0 )                        acc += BinaryMap_GetBit(image, x-1, y-1);
+
+  return acc;
+}
+
+static bool ArePointsEqual(Point p1, Point p2) {
+  return p1.x == p2.x && p1.y == p2.y;
+}
+
+static MinutiaType GetMinutiaType(int numberOfNeighbors) {
+  switch ( numberOfNeighbors ) {
+    case 1 : return RidgeEnd;
+    case 3: return Bifurcation;
+    default: return None;
+  }
+}
+
+static List GetPointsWithXNeighbors(int X, BinaryMap * image, List minutiae)
```

Comment:
Took me a while to work out what X meant, I initially thought it was to do with the width, might be better using N?